honeybadger.go

```diff
@@ -24,8 +49,54 @@ func (hb *HoneyBadger) sendBatch() {
 }
 
 // Create batch polling random transaction in queue
-func (hb *HoneyBadger) createBatch() {
-	panic("implement me w/ test case :-)")
+func (hb *HoneyBadger) createBatch() ([]Transaction, error) {
+	candidate, err := loadCand(hb, min(hb.b, hb.que.Len()))
+	if err != nil {
+		return nil, err
+	}
+
+	batch, cleanUp := selectRand(hb, candidate, int(hb.b/hb.n))
+	defer cleanUp()
+
+	return batch, nil
+}
+
+func loadCand(hb *HoneyBadger, candSize int) ([]Transaction, error) {
+	candidate := make([]Transaction, candSize)
+	var err error
+	for i := 0; i < candSize; i++ {
+		candidate[i], err = hb.que.Poll()
+
+		if err != nil {
+			return nil, err
+		}
+	}
+	return candidate, nil
+}
+
+func selectRand(hb *HoneyBadger, candidate []Transaction, selectSize int) ([]Transaction, func()) {
```

Comment:
1. plz change it into method of hb
2. how about `selectRandomTx`?
3. plz write comment about this function start with `selectRandomTx ~~`